src/lumigo_tracer/lumigo_utils.py

```diff
@@ -277,12 +279,25 @@ def _create_request_body(
 
 def establish_connection(host):
     try:
-        return http.client.HTTPSConnection(host, timeout=EDGE_TIMEOUT)
+        if is_valid_ip(host):
+            return http.client.HTTPSConnection(
+                host, timeout=EDGE_TIMEOUT, context=ssl._create_unverified_context()
+            )
+        else:
+            return http.client.HTTPSConnection(host, timeout=EDGE_TIMEOUT)
     except Exception as e:
         get_logger().exception(f"Could not establish connection to {host}", exc_info=e)
     return None
 
 
+def is_valid_ip(host: str) -> bool:
```

Comment:
the name is a bit confusing. This function only checks that the input looks like an IP, 
maybe we can call it "is_ip_address"?